pkg/scheduler/objects/application.go

```diff
@@ -2013,6 +2027,10 @@ func (sa *Application) LogAppSummary(rmID string) {
 	appSummary.ResourceUsage = nil
 }
 
+func (sa *Application) HasPlaceholderAllocation() bool {
+	return sa.hasPlaceholderAlloc.Load()
```

Comment:
See above: a mutex can be used. Setting the boolean happens inside a mutex locked application. This is the only read point.